Source/Cosmology/CosmoAMR.hpp

```diff
@@ -0,0 +1,45 @@
+/* GRChombo
+ * Copyright 2012 The GRChombo collaboration.
+ * Please refer to LICENSE in GRChombo's root directory.
+ */
+
+#ifndef CosmoAMR_HPP_
+#define CosmoAMR_HPP_
+
+#include "GRAMR.hpp"
+#include "PunctureTracker.hpp"
+
+/// A child of Chombo's AMR class to interface with tools which require
+/// access to the whole AMR hierarchy, and those of GRAMR
+/**
+ * This object inherits from GRAMR and adds tools required for BH spacetimes
+ */
+class CosmoAMR : public GRAMR
+{
+  private:
+    double m_K_mean;
+    double m_rho_mean;
+    double m_S_mean;
+
+  public:
+    PunctureTracker m_puncture_tracker;
+
+    CosmoAMR() {}
+
+    void set_interpolator(AMRInterpolator<Lagrange<4>> *a_interpolator) override
+    {
+        GRAMR::set_interpolator(a_interpolator);
+        m_puncture_tracker.set_interpolator(a_interpolator);
+    }
+
+    void set_K_mean(double a_K_mean) { m_K_mean = a_K_mean; }
```

Comment:
Maybe add a comment for each function as not completely obvious, or just one comment for all